compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -839,9 +840,24 @@ public void visit(BLangArrayLiteral arrayLiteral) {
     }
 
     public void visit(BLangRecordLiteral recordLiteral) {
-        recordLiteral.keyValuePairs.forEach(kv -> {
+        List<BLangRecordLiteral.BLangRecordKeyValue> keyValuePairs = recordLiteral.keyValuePairs;
+        keyValuePairs.forEach(kv -> {
             analyzeExpr(kv.valueExpr);
         });
+
+        Set<String> names = new HashSet<>();
+        for (BLangRecordLiteral.BLangRecordKeyValue recFieldDecl : keyValuePairs) {
+            BLangExpression keyExpr = recFieldDecl.getKey();
+            if (keyExpr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
```

Comment:
What I was referring to was the check at https://github.com/ballerina-platform/ballerina-lang/blob/567dddf9f26e4ca56580f077a01b01d0e62ac397/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L1713

The type checker verifies that only simple var refs are used in the literal. Therefore, you can just simply cast it to `BLangSimpleVarRef`. No need for the conditional check.

Reply:
It turned out map initializer literal could contain string keys.
Since both map and record literals are the same, we have to check and act accordingly.
Specially when `map m = { foo: "value", "foo": "value" }`.
Will update the code to handle to handle previous edge case.